src/terminal/char_grid/cursor.rs

```diff
@@ -24,10 +24,20 @@ pub struct Cursor {
     pub coords: Coords,
     pub style: Styles,
     pub text_style: Styles,
+    config: Config,
```

Comment:
> I think it would make sense to put the Config in an Rc, and have the Terminal and each CharGrid have a separate instance of it, but to pass it by reference to each function that needs it on any of the components of CharGrid.

Cool, I'll go ahead and take this approach then.